### DISCUSSION TITLE
Possible to run luarocks from within embedded Lua VM?

### DISCUSSION THREAD

**Discussion by maarek** [mlua-rs/mlua#377] [Q&A]
I'm exploring ideas for adding plugins to my rust application and one thought I had was to use mlua to embed the Lua VM and run provided lua scripts. What I am unsure of is whether I could pull in luarocks into the embedded vm and run them? I saw some documentation about building rocks from rust modules but not using rocks within the embedded lua.

**Comment by alerque** [comment]
Sure, you can do that just fine. I've set up [sile](https://github.com/sile-typesetter/sile) (develop branch) to run an entire Lua application including loading external LuaRocks (Lua or C modules). It also has the option to build with those external modules embeded in the Rust binary (again for both pure Lua and C modules) so it can be self contained. The latter involves a bit of calisthenics writing your own loaders, but the former is dead simple: just set the `package.path`/`package.cpath` to the correct values for where your LuaRocks are installed and away you go.
